addOns/pscanrulesAlpha/src/test/java/org/zaproxy/zap/extension/pscanrulesAlpha/SubResourceIntegrityAttributeScannerTest.java

```diff
@@ -0,0 +1,132 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import org.junit.Test;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+public class SubResourceIntegrityAttributeScannerTest
+        extends PassiveScannerTest<SubResourceIntegrityAttributeScanner> {
+
+    @Test
+    public void shouldNotRaiseAlertGivenIntegrityAttributeIsPresentInLinkElement()
+            throws HttpMalformedHeaderException {
+        // Given a HTML page with link element containing an integrity attribute
+        HttpMessage msg =
+                buildMessage(
+                        "<html><head><link rel=\"stylesheet\" href=\"https://site53.example.net/style.css\"\n"
+                                + "      integrity=\"sha384-+/M6kredJcxdsqkczBUjMLvqyHb1K/JThDXWsBVxMEeZHEaMKEOEct339VItX1zB\"\n"
+                                + "      crossorigin=\"anonymous\"></head><body></body></html>"); // from https://www.w3.org/TR/SRI/#use-casesexamples
+
+        // When the page is scanned
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then no alert should be raised
+        assertThat(alertsRaised.size(), equalTo(0));
+    }
+
+    @Test
+    public void shouldNotRaiseAlertGivenIntegrityAttributeIsPresentInScriptElement()
+            throws HttpMalformedHeaderException {
+        // Given a HTML page with link element containing an integrity attribute
+        HttpMessage msg =
+                buildMessage(
+                        "<html><head><script src=\"https://analytics-r-us.example.com/v1.0/include.js\"\n"
+                                + "        integrity=\"sha384-MBO5IDfYaE6c6Aao94oZrIOiC6CGiSN2n4QUbHNPhzk5Xhm0djZLQqTpL0HzTUxk\"\n"
+                                + "        crossorigin=\"anonymous\"></script></head><body></body></html>"); // from
+        // https://www.w3.org/TR/SRI/#use-casesexamples
+
+        // When the page is scanned
+        rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+        // Then no alert should be raised
+        assertThat(alertsRaised.size(), equalTo(0));
+    }
+
+    @Test
+    public void shouldRaiseAlertGivenIntegrityAttributeIsMissingForSupportedElement()
+            throws HttpMalformedHeaderException {
+        // Given a HTML page with a script element containing an integrity attribute
```

Comment:
There isn’t an attr here

Reply:
IMO these extra descriptions are unnecessary, the method already says what's being tested/asserted and it's more to maintain (or more likely, getting outdated/wrong).

Reply:
Well, I'm not a huge fan of highly arrange/act/assert with Given/When/Then.
It seems to me that it is a way to write Cucumber tests without the BDD methodology.
`// Given`, `// When` or `// Then` don't give much information.
If I want to keep the information, I would rather create a function like `given_a_message_with(XXX)`.

Reply:
No, they are just to delineate. The method name should give enough information (as it already does).